image_data_guide/02_structuring_data.ipynb

```diff
@@ -0,0 +1,198 @@
+    "|       +-- filename.jpg\n",
+    "|       +-- filename.jpg\n",
+    "+-- val\n",
+    "|   +-- class_A\n",
+    "|       +-- filename.jpg\n",
+    "|       +-- filename.jpg\n",
+    "|       +-- filename.jpg\n",
+    "|   +-- class_B\n",
+    "|       +-- filename.jpg\n",
+    "|       +-- filename.jpg\n",
+    "|       +-- filename.jpg\n",
+    "+-- test\n",
+    "|   +-- class_A\n",
+    "|       +-- filename.jpg\n",
+    "|       +-- filename.jpg\n",
+    "|       +-- filename.jpg\n",
+    "|   +-- class_B\n",
+    "|       +-- filename.jpg\n",
+    "|       +-- filename.jpg\n",
+    "|       +-- filename.jpg\n",
+    "```\n",
+    "You will notice that the COCO dataset does not come structured like above so we must use the annotation data to help restructure the folders of the COCO dataset so they match the pattern above. Once the new directory structures are created you can use your desired framework's data loading tool to gracefully load and define transformation for your image data. Many datasets may already be in this structure in which case you can skip this guide."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load Annotations Category Labels\n",
+    "The `sample_annos` and `category_labels` files were generated from the first notebook in this series `01_download_data.ipynb`. You will need to run that notebook before running the code here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open('pickled_data/sample_annos.pickle', 'rb') as f:\n",
+    "    sample_annos = pickle.load(f)\n",
+    "    \n",
+    "with open('pickled_data/category_labels.pickle', 'rb') as f:\n",
+    "    category_labels = pickle.load(f)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Divide the data into train, validation and test splits\n",
+    "We should divide our data into train, validation and test splits. A typical split ratio is 80/10/10. Our image classification algorithm will train on the first 80% (training) and evaluate its performance at each epoch with the next 10% (validation) and we'll give our model's final accuracy results using the last 10% (test). It's important that before we split the data we make sure to shuffle it randomly so that class distribution among splits is roughly proportional."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "np.random.seed(0)\n",
+    "image_ids = sorted(list(sample_annos.keys()))\n",
+    "np.random.shuffle(image_ids)\n",
+    "first_80 = int(len(image_ids) * 0.8)\n",
+    "next_10 = int(len(image_ids) * 0.9)\n",
+    "train_ids, val_ids, test_ids = np.split(image_ids, [first_80, next_10])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Make new folder structure and copy image files"
```

Comment:
Explain why they would want and/or need to do this.

Reply:
Added explanation.